django/template/defaultfilters.py

```diff
@@ -57,6 +58,26 @@ def _dec(*args, **kwargs):
 # STRINGS         #
 ###################
 
+@stringfilter
+def autoescape_aware(func):
+    """
+    Decorator for filters which should automatically escape text based
+    on whether or not autoescape is True.
+    """
+    def _dec(*args, **kwargs):
+        try:
+            autoescape = kwargs['autoescape']
+        except KeyError:
+            autoescape = signature(func).parameters['autoescape'].default
+        if (autoescape):
+            return conditional_escape(args[0])
+        return func(*args, **kwargs)
+
+    _dec._decorated_function = getattr(func, '_decorated_function', func)
+
+    return wraps(func)(_dec)
```

Comment:
Please use 

```python
  @wraps(func)
  def _dec(*args, **kwargs):
    ...
  return _dec
```
and consider naming `_dec` to something else, `wrapper` would be great and also don't use `_` before your names that's a convention only used for protected members in python. Local variables are not visible to outer scope by default.